deploy/Managing_Keystores_And_Truststores.md

```diff
@@ -0,0 +1,73 @@
+it is often required to customize the default, WSO2 product keystore and truststore in a production grade deployment.
+
+The most appropriate approach for introducing a custom keystore or a truststore to a container based deployment is by using
+volume mounting.
+
+1. Prepare the custom Java keystore.
+
+   Please follow the official guidelines for [creating a new keystore](https://docs.wso2.com/display/ADMIN44x/Creating+New+Keystores#CreatingNewKeystores-Creatinganewkeystore)
+   or [adding a Certificate Authority signed certificate to an existing keystore](https://docs.wso2.com/display/ADMIN44x/Creating+New+Keystores#CreatingNewKeystores-ca_certificateAddingCA-signedcertificatestokeystores)
+   as per your preference.
+   
+   To import certificates to a truststore, follow the official [documentation](https://docs.wso2.com/display/ADMIN44x/Creating+New+Keystores#CreatingNewKeystores-Step3:Importingcertificatestothetruststore).
+   
+2. Add the keystore to the product container via a volume mount depending on the container platform.
+
+   We recommend you to use the volume mount point allocated for non-configuration resources in WSO2 product containers
+   (i.e. `/home/wso2carbon/wso2-artifact-volume`) for this purpose.
+   
+   In Carbon Kernel **version 4** based products, the default `wso2carbon.jks` keystore is located at the
+   `<WSO2_PRODUCT_HOME>/repository/resources/security` directory. Thus, you may mount the volume containing the
+   custom keystore to `/home/wso2carbon/wso2-artifact-volume/repository/resources/security` directory.
+   
+   In Carbon Kernel **version 5** based products, the default `wso2carbon.jks` keystore is located at the
+   `<WSO2_PRODUCT_HOME>/resources/security` directory. Thus, you may mount the volume containing the custom keystore
+   to `/home/wso2carbon/wso2-artifact-volume/resources/security` directory.
+   
+3. Add the truststore to the product container via a volume mount depending on the container platform.
+
+   We recommend you to use the volume mount point allocated for non-configuration resources in WSO2 product containers
+   (i.e. `/home/wso2carbon/wso2-artifact-volume`) for this purpose.
+   
+   In Carbon Kernel **version 4** based products, the default `client-truststore.jks` truststore is located at the
+   `<WSO2_PRODUCT_HOME>/repository/resources/security` directory. Thus, you may mount the volume containing the
+   custom keystore to `/home/wso2carbon/wso2-artifact-volume/repository/resources/security` directory.
+   
+   In Carbon Kernel **version 5** based products, the default `client-truststore.jks` truststore is located at the
+   `<WSO2_PRODUCT_HOME>/resources/security` directory. Thus, you may mount the volume containing the custom keystore
+   to `/home/wso2carbon/wso2-artifact-volume/resources/security` directory.
+    
+   > In Kubernetes based WSO2 product deployment, it is advisable that you use a ConfigMap resource to add the custom keystore
```

Comment:
Would it be better if we mention the Kubernetes version that supports this approach? AFAIR it is `1.10.0` right?.

Reply:
@ThilinaManamgoda +1. Will add.

Reply:
Added in https://github.com/wso2/container-guide/pull/15/files#diff-9bf3730dcb649a313efa880c13afc382R69.